### ISSUE TITLE
Tests fail if file with type-only import is loaded

### ISSUE THREAD

**Issue by Simon-Hayden-iteratec** [nrwl/nx#7845]
## Current Behavior

Executing tests fail if type-only imports are used. Tested with Nx/Angular 12 and after updating to 13. Building and serving such apps works as intended - only testing is affected.

## Expected Behavior

Type-only imports must be interpreted correctly during testing. To my knowledge this is not a regression, but I havn't been using Nx for long yet.

## Steps to Reproduce

I bootstrapped a new project using Angular as base, but I think this might also occur with other types of projects:

https://github.com/Simon-Hayden-iteratec/type-only-imports-test

What I did:
1. Create a new project using `npx create-nx-workspace@latest type-only-imports-test` and Angular as base
2. Create a new Angular service, which wants to inject "Request" from express (with a token, hence create a type-only import for Request).
3. Try to execute the tests for the service, they will fail (see Failure Logs)

We need this, because we are using Angular's Server Side Rendering and therefore (sometimes) have to manipulate the response or read from the request. However, since no "express" code should land in the compilation, we need the type-only import.

I tried the same setup, but using `ng` to generate a new application and service, which worked without issue. So something of the Nx configurations for the test environment breaks it.

### Failure Logs

```
 FAIL   angular  apps/angular/src/app/app.service.spec.ts
  ● AppService › should be created

    ReferenceError: Request is not defined

      1 | import { Inject, Injectable, Optional } from '@angular/core';
    > 2 | import type { Request } from 'express';
        |               ^
      3 | import { REQUEST } from './app.tokens';
      4 |
      5 | @Injectable({
```

### Environment

```
  Node : 14.17.6
  OS   : linux x64
  npm  : 6.14.15
  
  nx : 13.2.2
  @nrwl/angular : 13.2.2
  @nrwl/cli : 13.2.2
  @nrwl/cypress : 13.2.2
  @nrwl/devkit : 13.2.2
  @nrwl/eslint-plugin-nx : 13.2.2
  @nrwl/express : undefined
  @nrwl/jest : 13.2.2
  @nrwl/linter : 13.2.2
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : undefined
  @nrwl/nx-cloud : undefined
  @nrwl/react : undefined
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.2.2
  @nrwl/web : undefined
  @nrwl/workspace : 13.2.2
  @nrwl/storybook : 13.2.2
  @nrwl/gatsby : undefined
  typescript : 4.4.4
  rxjs : 7.4.0
  ---------------------------------------
  Community plugins:
         @angular/animations: 13.0.2
         @angular/common: 13.0.2
         @angular/compiler: 13.0.2
         @angular/core: 13.0.2
         @angular/forms: 13.0.2
         @angular/platform-browser: 13.0.2
         @angular/platform-browser-dynamic: 13.0.2
         @angular/router: 13.0.2
         @angular-devkit/build-angular: 13.0.3
         @angular/cli: 13.0.3
         @angular/compiler-cli: 13.0.2
         @angular/language-service: 13.0.2
```

**Comment by leosvelperez** [comment]
Thanks for reporting this!

I wonder what did you try to reproduce the issue with the Angular CLI that works. I put together a repo https://github.com/leosvelperez/ng-jest-type-issue with just the Angular CLI and Jest and the same error pops up. Maybe you tried the default Angular CLI test setup which uses Karma instead of Jest 🤔 

Anyways, given the same problem occurs with an Angular CLI workspace and Jest, I don't think the issue has anything to do with the Nx setup. The `jest-preset-angular` repository might be a good place to raise this issue.

**Comment by Simon-Hayden-iteratec** [comment]
Yes, I did use the default Angular configuration with Karma. Thanks for double checking!

I will then raise the issue to the `jest-preset-angular` team. The reason I posted this here first, was because jest had this issue already fixed a year and a half ago: https://github.com/facebook/jest/issues/9611

I will close the issue for now and create a new one if there are some nx specific issues.
